packages/sui-hoc/src/withIntersectionObserver.js

```diff
@@ -0,0 +1,48 @@
+/* global IntersectionObserver */
+import React, {Component} from 'react'
+
+if (typeof window !== 'undefined' && window.document) {
```

Comment:
What about using a dynamic import only if the browser doesn't have support for it. So the check would be done in the componentDidMount, which is only executed in the client.

Reply:
With this code 
```
if (typeof window !== 'undefined' && window.document) {
  require('intersection-observer')
}
```
we assure we only load the library/pollyfill in the client. 

For what I've seen in other implementations in Schibsted, the general criteria is to load always the pollyfill and let that code to check if the specific feature is available and apply proper changes if not

What exactly is your suggestion here?

Reply:
using that IF you will end with the polyfill in your bundler. You have to use the `import()` statement to have lazy loading of this polyfill.